appveyor.yml

```diff
@@ -59,8 +60,41 @@ test_script:
 - ps: |
     Write-Host "Tests are running..."
     Write-Host $pwd
-    & .\cibuild.cmd -test -integrationTest
-    if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
+    $testErrorsTotal = 0
+    $testRunCount = $env:TEST_RUN_COUNT
+    for ($testRun=1; $testRun -le $testRunCount; $testRun++)
+    {
+        Write-Host "[INFO]: Test Run ${testRun}/${testRunCount}, ${testErrorsTotal} test(s) failed before"
+
+        & .\cibuild.cmd -integrationTest
+        & .\cibuild.cmd -test
+
+        Get-ChildItem -recurse artifacts\bin\tests\TestResult.xml `
+        | ForEach-Object {
+            $resultFilePath = "$_"
+            Get-Content $resultFilePath `
+            | Select-String -Pattern '<test-run id="[^"]*"\s+testcasecount="(\d+)"\s+result="([^"]*)"\s+total="(\d+)"\s+passed="(\d+)"\s+failed="(\d+)".*' `
+            | foreach {
+                $result = $_.Matches.Groups
+                #$result[0].Value
+                $total  = $result[1].Value
+                $passed = $result[4].Value
+                $failed = $result[5].Value
+                $notPassed = $total - $passed
+                $testErrors = $failed
+                $testName = $resultFilePath.Split('\\')[-3]
+                $artifactName = "TestResult#${testRun}.${testErrors}err.${testName}.xml"
+                Push-AppveyorArtifact $resultFilePath -FileName $artifactName
```

Comment:
Do we need to publish these?

Reply:
Yes, because they contain the full truth ;-). Instead of these, I have omitted the `.log` of successful tests.

Reply:
Is there a benefit in publishing them for successful tests though?

Reply:
We can omit them when the tests pass stably. Until then, they might contain e.g. printf debug output for comparison with failing tests.
In addition, we have a test which is not executed although it should: #7695. This is visible in the `TestResult.xml`, too.